src/NLog/Config/XmlLoggingConfiguration.cs

```diff
@@ -82,21 +84,23 @@ private ConfigurationItemFactory ConfigurationItemFactory
         /// Initializes a new instance of the <see cref="XmlLoggingConfiguration" /> class.
         /// </summary>
         /// <param name="fileName">Configuration file to be read.</param>
-        public XmlLoggingConfiguration(string fileName)
-        {
-            using (XmlReader reader = CreateFileReader(fileName))
-            {
-                this.Initialize(reader, fileName, false);
-            }
-        }
+        /// <param name="customLogFactory">The custom <see cref="LogFactory"/> to which to apply any applicable configuration values,
+        /// or <c>null</c> to use the global <see cref="LogFactory"/> (via <see cref="LogManager"/>).</param>
+        public XmlLoggingConfiguration(string fileName, LogFactory customLogFactory = null) 
```

Comment:
Looks good but optional parameters aren't binary compatible. So we need extra ctors and should not alter the current ones.